### PR TITLE
[DX] Added info about security.current_user_provider

### PR DESCRIPTION
**Pull request by Nyholm**
| Q | A |
| --- | --- |
| Doc fix? | no |
| New docs? | new feature: https://github.com/symfony/symfony/pull/14407 |
| Applies to | 2.8 and above |
| Fixed tickets | none |

### CODE REVIEW

File: book/security.rst [hunk 1]
```diff
@@ -1043,6 +1043,17 @@ if your User object has a ``getFirstName()`` method, you could use that::
...
     }
 
+.. versionadded:: 2.8
+     The ``security.current_user_provider`` service was introduced in Symfony 2.8. It makes it 
+     easier to retrieve the current logged in user from a service. 
+     
+    .. code-block:: php
+
+        $user = $this->container->get('security.current_user_provider')->getUser();
+        if ($user === null) {
```
Comment (javiereguiluz) [inline book/security.rst]: I have two comments about this line:

1. Should we use in the docs the same conditional style as in the code (`if (null === $user)`)?
2. In the docs we explain that in Symfony applications, even anonymous users are technically logged in. Not only that, but they are even authenticated ( `IS_AUTHENTICATED_ANONYMOUSLY`). That's why I find strange that the user can be `null` here. I expect to get an anonymous user.

Comment (WouterJ) [inline book/security.rst]: (1) Yes, it should be switched indeed.
(2) I agree with @javiereguiluz here, I've left a comment in the code PR.

**Comment by WouterJ** [comment]
We often try to keep the versionadded directives as small as possible. Can you please change this to something like:

```rst
You can use the ``security.current_user_provider`` service to
get the current user::

    $user = $this->get('security.current_user_provider')->getUser();

.. versionadded:: 2.8
    The ``security.current_user_provider`` service was introduced in Symfony 2.8.
```

**Comment by Nyholm** [comment]
Thank you @WouterJ and @javiereguiluz for your comments. I will update accordingly. 

I know it's technically incorrect to discard the anonymous user. But it is actually what developers are doing when we call `$user = $this->get('security.current_user_provider')->getUser();`. Lets have that discussion at the PR with the code.

**Pull request closed without merging**